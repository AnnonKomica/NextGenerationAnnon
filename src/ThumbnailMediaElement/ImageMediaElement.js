// @flow
import React from "react";
import ClassNameUtil from "../utils/ClassNameUtil";
import ThumbnailMediaElement from "./ThumbnailMediaElement";

export default class ImageMediaElement extends ThumbnailMediaElement {
    _getRawMediaElement(): ?Object {
        const {
            showRaw,
            detailLoaded
        } = this.state;

        if (!showRaw) {
            return null;
        }

        const rawMediaClass = ClassNameUtil.classNames(["thumbnail-media__raw-media--image"], {
            "thumbnail-media__raw-media--loading": !detailLoaded,
        });

        return (
            <img
                className={rawMediaClass}
                src={this.props.rawMediaSrc}
                onLoad={this._handleRawElementLoad}
                onClick={this._handleRawElementClick}
            />
        );
    }
}
