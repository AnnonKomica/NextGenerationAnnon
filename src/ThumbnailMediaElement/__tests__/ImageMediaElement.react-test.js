import React from "react";
import renderer from "react-test-renderer";

import ImageMediaElement from "../ImageMediaElement";

// eslint-disable-next-line
test("ImageMediaElement DOM should render correctly", () => {
    const component = renderer.create(
        <ImageMediaElement
            thumbnailSrc="thumbnailSrc"
            rawMediaSrc="rawMediaSrc"
        />
    );

    let tree = component.toJSON();

    // eslint-disable-next-line
    expect(tree).toMatchSnapshot();

    tree.children[0].props.onClick();
    tree = component.toJSON();

    // eslint-disable-next-line
    expect(tree).toMatchSnapshot();

    tree.children[1].props.onLoad();
    tree = component.toJSON();

    // eslint-disable-next-line
    expect(tree).toMatchSnapshot();


    tree.children[1].props.onClick();
    tree = component.toJSON();

    // eslint-disable-next-line
    expect(tree).toMatchSnapshot();
});
