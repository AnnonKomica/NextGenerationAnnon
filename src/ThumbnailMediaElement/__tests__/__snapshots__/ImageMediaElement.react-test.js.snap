exports[`test ImageMediaElement DOM should render correctly 1`] = `
<div>
  <img
    className="thumbnail-media__thumbnail"
    onClick={[Function]}
    src="thumbnailSrc" />
</div>
`;

exports[`test ImageMediaElement DOM should render correctly 2`] = `
<div>
  <img
    className="thumbnail-media__thumbnail thumbnail-media__thumbnail--raw-loading"
    onClick={[Function]}
    src="thumbnailSrc" />
  <img
    className="thumbnail-media__raw-media--image thumbnail-media__raw-media--loading"
    onClick={[Function]}
    onLoad={[Function]}
    src="rawMediaSrc" />
</div>
`;

exports[`test ImageMediaElement DOM should render correctly 3`] = `
<div>
  <img
    className="thumbnail-media__thumbnail thumbnail-media__thumbnail--show-raw"
    onClick={[Function]}
    src="thumbnailSrc" />
  <img
    className="thumbnail-media__raw-media--image"
    onClick={[Function]}
    onLoad={[Function]}
    src="rawMediaSrc" />
</div>
`;

exports[`test ImageMediaElement DOM should render correctly 4`] = `
<div>
  <img
    className="thumbnail-media__thumbnail"
    onClick={[Function]}
    src="thumbnailSrc" />
</div>
`;
